test/integration/dynamic-routing/test/index.test.js

```diff
@@ -84,6 +84,12 @@ function runTests(dev) {
     expect(url).toBe('?fromHome=true')
   })
 
+  it('should not have any console warnings', async () => {
+    const browser = await webdriver(appPort, '/')
+    const caughtWarns = await browser.eval(`window.caughtWarns`)
+    expect(caughtWarns).toEqual([])
+  })
```

Comment:
Looks good! Can we also test to make sure there's no warnings after clicking on a link to a dynamic route?

Reply:
Done! I ended up adding a new page without an empty `getInitialProps` function since that was actually throwing a dev warning.